### PR TITLE
Update SortingAlgorithm.swift

### PR DESCRIPTION
**Pull request by darhonbek**
- Resolve complexity issue in merge sort. Bring down from O(n^2) to O(n).

### CODE REVIEW

File: Sources/DataStructuresAndAlgorithms/Algorithms/SortingAlgorithm.swift [hunk 1]
```diff
@@ -131,25 +131,27 @@ struct SortingAlgorithm {
     }
 
     private func merge<T: Comparable>(n1: [T], n2: [T]) -> [T] {
-        var n1 = n1
-        var n2 = n2
+        let size1 = n1.count
+        let size2 = n2.count
```
Comment (sukhrobkhakimov) [inline Sources/DataStructuresAndAlgorithms/Algorithms/SortingAlgorithm.swift:135]: @darhonbek I don't want to use `count` for performance reasons. When you have a lot of items, it starts making a difference.

File: Sources/DataStructuresAndAlgorithms/Algorithms/SortingAlgorithm.swift [hunk 2]
```diff
@@ -131,25 +131,27 @@ struct SortingAlgorithm {
     }
 
     private func merge<T: Comparable>(n1: [T], n2: [T]) -> [T] {
```
Comment (sukhrobkhakimov) [inline Sources/DataStructuresAndAlgorithms/Algorithms/SortingAlgorithm.swift:133]: This is not a main function for `mergeSort`.

**Review by sukhrobkhakimov — COMMENTED** [review]
Have you run it? It sorts in an ascending order. [https://ibb.co/vs5nXHh](url)

File: Sources/DataStructuresAndAlgorithms/Algorithms/SortingAlgorithm.swift [hunk 3]
```diff
@@ -131,25 +131,27 @@ struct SortingAlgorithm {
...
-        var n1 = n1
-        var n2 = n2
+        let size1 = n1.count
```
Comment (sukhrobkhakimov) [inline Sources/DataStructuresAndAlgorithms/Algorithms/SortingAlgorithm.swift]: Rename it to `s1` for consistency reasons

File: Sources/DataStructuresAndAlgorithms/Algorithms/SortingAlgorithm.swift (hunk 1 above)
Comment (sukhrobkhakimov) [inline Sources/DataStructuresAndAlgorithms/Algorithms/SortingAlgorithm.swift]: Rename it to `s2` for consistency reasons

**Review by sukhrobkhakimov — COMMENTED** [review]
I tested the speed and yours is better. Can you please add some minor changes so that I can merge it? Thanks.

**Comment by darhonbek** [comment]
> Have you run it? It sorts in an ascending order. [https://ibb.co/vs5nXHh](url)

Sorry, turns out I was running my re-write. No issues with sorting order. My bad)

File: Sources/DataStructuresAndAlgorithms/Algorithms/SortingAlgorithm.swift (hunk 3 above)
Comment (darhonbek) [reply Sources/DataStructuresAndAlgorithms/Algorithms/SortingAlgorithm.swift]: Done!

File: Sources/DataStructuresAndAlgorithms/Algorithms/SortingAlgorithm.swift (hunk 1 above)
Comment (darhonbek) [reply Sources/DataStructuresAndAlgorithms/Algorithms/SortingAlgorithm.swift]: Done!

**Review by sukhrobkhakimov — APPROVED** [review]

**Pull request merged**